sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1136,6 +1150,64 @@ Exceptions:
   `property::graph::updatable` was not set when the executable graph was created.
 * Throws with error code `invalid` if any node in `nodes` is not part of the
   graph.
+
+|
+[source, c++]
+----
+void
+update(const command_graph<graph_state::modifiable>& graph);
+----
+
+|Updates the executable graph node inputs & outputs from a topologically
+identical modifiable graph. A topologically identical graph is one with the
+same structure of nodes and edges, and the nodes added in the same order to
+both graphs. Equivalent nodes in topologically identical graphs each have the
+same command, targeting the same device. There is the additional limitation that
+to update an executable graph, every node in the graph must be a kernel command.
+
+During an update, the only characteristics that can differ between two topologically
+identical graphs are the arguments and the execution ND-Range of the kernel 
+nodes. For example, the graph may capture different values for the USM pointers 
+or accessors used in the graph. It is these kernels arguments in `graph` that 
+constitute the inputs & outputs to update to.
+
+Differences in the following characteristics between two graphs during an
+update results in undefined behavior:
+
+* Modifying the {sycl-kernel-function}[kernel function] of a kernel node.
+
+The effects of the update will be visible on the next submission of the
+executable graph without the need for additional user synchronization.
```

Comment:
We have the following wording in the other `update()` descriptions and I think it would be useful to add here too

> The implementation may perform a blocking wait during this call on any in-flight executions of that same graph if required by the backend.